homeassistant/components/modbus/sensor.py

```diff
@@ -271,7 +276,7 @@ def _swap_registers(self, registers):
             # convert [12][34] --> [21][43]
             for i, register in enumerate(registers):
                 registers[i] = int.from_bytes(
-                    register.to_bytes(2, byteorder="little"),
+                    register.to_bytes(self._register_size, byteorder="little"),
```

Comment:
that looks ok, but the lines that follow have not been changed, that was not my expectation.....but lets wait and see the tests.

Reply:
Code only work with "2"

Reply:
meanwhile I know you miss to correct the remaining part of swap.